Sources/MongoSwift/SDAM.swift

```diff
@@ -22,6 +25,7 @@ public struct ConnectionId: Equatable {
     internal init(_ hostAndPort: String = "localhost:27017") {
         let parts = hostAndPort.split(separator: ":")
         self.host = String(parts[0])
+        // swiftlint:disable:next force_unwrapping - should be valid UInt16 unless server response malformed.
```

Comment:
Could potentially mark this as `throws` and throw an `invalidResponse` error. I'm not sure how common it would be for the response to be malformed or what the consequences of this initializer being throwable are, though.

Reply:
yeah, I considered erroring here as well. I ended up not doing so because everywhere this is called from doesn't throw right now, so there would have been a lot of additional refactoring needed. this is used in the context of SDAM monitoring, so it's also not clear what the behavior should be if parsing isMaster fails. do we just skip publishing the event? do we publish it with a incorrect or missing `ConnectionId`? I opened [SWIFT-314](https://jira.mongodb.org/browse/SWIFT-314) about this.